src/common/cockpitpipe.c

```diff
@@ -1116,11 +1116,18 @@ cockpit_pipe_connect (const gchar *name,
   else
     {
       if (!g_unix_set_fd_nonblocking (sock, TRUE, NULL))
-        g_return_val_if_reached (NULL);
+        {
+          close (sock);
+          g_return_val_if_reached (NULL);
```

Comment:
g_return_xxx() and g_critical() are treated like a soft-kernel Ooops. As in the state of the process is undefined and unexpected after that point. We'll try to keep running, but especially regards with respect to memory leaks, all bets are off.